compiler/rustc_abi/src/lib.rs

```diff
@@ -221,6 +221,19 @@ impl ReprOptions {
 /// * Cranelift stores the base-2 log of the lane count in a 4 bit integer.
 pub const MAX_SIMD_LANES: u64 = 1 << 0xF;
 
+/// Informations relative to a specific address space.
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct AddressSpaceInfo {
+    /// The size of the bitwise representation of the pointer.
+    pointer_size: Size,
+    /// The ABI alignment requirements for pointers in this address space.
+    pointer_align: AbiAlign,
+    /// The preferred alignment specification for pointers in this address space.
+    pointer_preferred_align: AbiAlign,
+    /// The size of the index that used for address calculations on pointers in this address space.
+    pointer_index: Size,
+}
```

Comment:
```suggestion
    /// Pointers into this address space contain extra metadata
    /// FIXME(workingjubilee): Consider adequately reflecting this in the compiler?
    _is_fat: bool,
}
```

Reply:
[58055f8](https://github.com/rust-lang/rust/pull/143182/commits/58055f8eafd52dd001eda3a8e5f57d0aa68368df) adds the field.